src/collections/componentSet.ts

```diff
@@ -76,11 +79,11 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
     this.logger = Logger.childFromRoot(this.constructor.name);
 
     for (const component of components) {
-      let asDeletion = false;
+      let destructiveType = this.destructiveChangesType;
```

Comment:
Since this is an "if-else" assignment, a turnery operation can be used here.
```
const destructiveType = (component instanceof SourceComponent)
  ? component.getDestructiveChangesType(),
  : this.destructiveChangesType;
```